### PR TITLE
Micrometer instrumentation, part 3

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Continuation of https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/5001

This PR includes the last missing instruments: `LongTaskTimer` and `Meter`.

### CODE REVIEW

**Review by anuraaga — APPROVED** [review]

File: instrumentation/micrometer/micrometer-1.5/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/AsyncInstrumentRegistry.java [hunk 1]
```diff
@@ -21,10 +21,17 @@
...
+// TODO: refactor this class, there's too much copy-paste here
 final class AsyncInstrumentRegistry {
 
   private final Meter meter;
 
+  // we're always locking lock on the whole instrument map; the add/remove methods aren't called
+  // that often, so it's probably better to opt for correctness in that case - there is a small
+  // window between removing a single measurement and removing the whole instrument (if it has no
+  // more measurements) when potentially a new measurement could be added; a ConcurrentHashMap
+  // wouldn't be enough in this case
```
Comment (trask) [inline instrumentation/micrometer/micrometer-1.5/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/AsyncInstrumentRegistry.java:33]: thx

File: instrumentation/micrometer/micrometer-1.5/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/LongTaskTimerTest.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+                metric ->
+                    assertThat(metric)
+                        .hasDescription("This is a test long task timer")
+                        .hasUnit("ms")
+                        .hasDoubleHistogram()
+                        .points()
+                        .satisfiesExactly(
+                            point ->
+                                assertThat(point)
+                                    /* .hasSumGreaterThan(100) */
```
Comment (trask) [inline instrumentation/micrometer/micrometer-1.5/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/LongTaskTimerTest.java:84]: can we keep this? (i see the sleep 100 above)

File: instrumentation/micrometer/micrometer-1.5/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/LongTaskTimerTest.java [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+                        .satisfiesExactly(
+                            point ->
+                                assertThat(point)
+                                    .hasValue(0)
+                                    .attributes()
+                                    .containsOnly(attributeEntry("tag", "value")))));
+
+    // when
+    TimeUnit.MILLISECONDS.sleep(100);
+    sample.stop();
```
Comment (trask) [inline instrumentation/micrometer/micrometer-1.5/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/LongTaskTimerTest.java:128]: did you mean to assign sample to `timer.start()` above?

File: instrumentation/micrometer/micrometer-1.5/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/Bridging.java [hunk 4]
```diff
@@ -38,5 +39,13 @@ static String baseUnit(Meter.Id id) {
     return baseUnit == null ? "1" : baseUnit;
   }
 
+  static String statisticInstrumentName(Meter.Id id, Statistic statistic) {
+    String prefix = id.getName() + ".";
+    // use "total_time" instead of "total" to avoid clashing with Statistic.TOTAL
```
Comment (trask) [inline instrumentation/micrometer/micrometer-1.5/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/Bridging.java:44]: 👍

**Review by trask — APPROVED** [review]

File: instrumentation/micrometer/micrometer-1.5/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/LongTaskTimerTest.java (hunk 2 above)
Comment (mateuszrzeszutek) [reply instrumentation/micrometer/micrometer-1.5/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/LongTaskTimerTest.java:84]: Oh, that method is already available in 1.10 sdk-testing - I'll uncomment these. Thanks!

File: instrumentation/micrometer/micrometer-1.5/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/LongTaskTimerTest.java (hunk 3 above)
Comment (mateuszrzeszutek) [reply instrumentation/micrometer/micrometer-1.5/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/LongTaskTimerTest.java:128]: Nice catch! I've noticed that `Sample#stop()` was a bit broken thanks to that - fixed that and added one more test. Thanks!

**Pull request merged**